internal/common/crypto_policy.go

```diff
@@ -0,0 +1,126 @@
+package common
+
+import (
+	"context"
+	"crypto/tls"
+	"errors"
+	"fmt"
+
+	"github.com/go-logr/logr"
+	ocpv1 "github.com/openshift/api/config/v1"
+	"github.com/openshift/library-go/pkg/crypto"
+	"kubevirt.io/ssp-operator/api/v1beta1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type SSPTLSOptions struct {
+	MinTLSVersion      string
+	OpenSSLCipherNames []string
+}
+
+func (s *SSPTLSOptions) IsEmpty() bool {
+	return len(s.OpenSSLCipherNames) == 0 && s.MinTLSVersion == ""
+}
+
+func (s *SSPTLSOptions) MinTLSVersionId() (minTlsVersionId uint16) {
+	switch s.MinTLSVersion {
+	case "1.0":
+		return tls.VersionTLS10
+	case "1.1":
+		return tls.VersionTLS11
+	case "1.2":
+		return tls.VersionTLS12
+	case "1.3":
+		return tls.VersionTLS13
+	default:
+		return 0
+	}
+}
+
+func NewSSPTLSOptions(tlsSecurityProfile *ocpv1.TLSSecurityProfile, logger *logr.Logger) (options *SSPTLSOptions) {
+	ciphers, tlsProfile := selectCipherSuitesAndMinTLSVersion(tlsSecurityProfile)
+
+	if logger != nil {
+		logger.Info("Got Ciphers and tlsProfile:", "ciphers: ", ciphers, "tlsProfile: ", tlsProfile)
+	}
+
+	return &SSPTLSOptions{
+		MinTLSVersion:      tlsVersionToHumanReadable(tlsProfile),
+		OpenSSLCipherNames: ciphers,
+	}
+}
+
+func (s *SSPTLSOptions) CipherIDs() (cipherSuites []uint16) {
+	for _, cipherName := range crypto.OpenSSLToIANACipherSuites(s.OpenSSLCipherNames) {
+		for _, knownCipher := range tls.CipherSuites() {
+			if knownCipher.Name == cipherName {
+				cipherSuites = append(cipherSuites, knownCipher.ID)
+			}
+		}
+	}
+	return
+}
+
+func emptyTLSProfile() *SSPTLSOptions {
+	return &SSPTLSOptions{"", []string{}}
+}
+
+var (
+	setupLog = ctrl.Log.WithName("setup")
+)
+
+func GetCipherSuitesAndMinTLSVersion() (*SSPTLSOptions, error) {
+	restConfig := ctrl.GetConfigOrDie()
+	apiReader, err := client.New(restConfig, client.Options{Scheme: Scheme})
+	if err != nil {
+		return emptyTLSProfile(), err
+	}
+
+	namespace, err := GetOperatorNamespace(setupLog)
+	if err != nil {
+		return emptyTLSProfile(), err
+	}
+
+	var sspList v1beta1.SSPList
+	if err := apiReader.List(context.TODO(), &sspList, &client.ListOptions{Namespace: namespace}); err != nil {
+		return emptyTLSProfile(), fmt.Errorf("error getCipherSuitesAndMinTLSVersion: %w", err)
```

Comment:
Do we need to wrap the error? Does it have a benefit?

Reply:
I would like to because it is a public method, what do you think?

Reply:
Wrapping error is ok, but the text should provide more information about the function that failed. In this case something like: `"failed listing SSP resources: %w"`.
The calling function of `GetCipherSuitesAndMinTLSVersion()` will know that the failure was in that function, so text `"error getCipherSuitesAndMinTLSVersion"` is useless.

Reply:
Ok, but then the other errors which can be returned should be wrapped too with a more explanative error message. You agree?

Reply:
Yes. If we want to wrap errors, then it makes sens to wrap all of them. Unless it is obvious where the error happened without wrapping. But in this case there are multiple calls to the `apiReader` and they could fail with simillar error messages, so wrapping is useful.